src/MICore/UnixUtilities.cs

```diff
@@ -160,5 +163,28 @@ internal static bool IsProcessRunning(int processId)
             // if there is no process with the ID specified.
             return UnixNativeMethods.GetPGid(processId) >= 0;
         }
+
+        public static bool IsBinarySigned(string filePath)
```

Comment:
We are only checking once, so not sure I follow why we would cache it.

On top of that, signing is a "dynamic" property. All the user has to do is execute

`$ codesign -f -s  "<cert name>"  <debugger path>`